### PR TITLE
Week 3 Walkthrough Part 2

### PR DESCRIPTION
**Pull request by dtalton3**
Just completed the fetch walkthrough for the week 3 project. 

- Modified the javascript, and html files to use a fetch request instead of hardcoded facts for the fun fact generator.
- pom.xml needed the dependency with the servlet, so I added that.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,14 +19,40 @@
...
+    facts.add("I am interested in videography and photography");
+    facts.add("My dream job is to work with the team that developed YouTube, or create my own rivalling service.");
+    facts.add("I would have been in Bellevue, Washington if not for COVID-19");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Desmond!</h1>");
+
```
Comment (octopus1000) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Please remove unnecessary empty lines.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -19,14 +19,40 @@
...
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Desmond!</h1>");
+    
+    String json = convertToJsonUsingGson(facts);
+    response.setContentType("application/json");
+
```
Comment (octopus1000) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Empty line

**Review by octopus1000 — COMMENTED** [review]
Were you intended to include any js changes?

**Comment by dtalton3** [comment]
> Were you intended to include any js changes?

Yes. I thought I did. I'll include it with my resolved commit.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (dtalton3) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Will do.

**Review by addarsh-google — APPROVED** [review]

**Pull request merged**